src/sched/spin_impl.cc

```diff
@@ -0,0 +1,42 @@
+#include "cris/core/sched/spin_impl.h"
+
+#include <cstddef>
+
+#define _CR_ASM_NOP asm volatile("nop")
+
+// Cannot use `do { } while (0)` here because there would be huge differences between -O0 and -O1.
+#define _CR_2X(statement) \
+    {                     \
+        statement;        \
+        statement;        \
+    }
+
+#define _CR_2X_NOP _CR_2X(_CR_ASM_NOP)
+#define _CR_4X_NOP _CR_2X(_CR_2X_NOP)
+#define _CR_8X_NOP _CR_2X(_CR_4X_NOP)
+#define _CR_16X_NOP _CR_2X(_CR_8X_NOP)
+#define _CR_32X_NOP _CR_2X(_CR_16X_NOP)
+
+namespace cris::core::impl {
+
+void SpinForAbout1us() {
```

Comment:
This should be manually inlined by macro, not as function.
If it's in DLL, which cannot be inlined by LTO, it can be quite expensive.

Reply:
And this piece of code itself cannot control how it's built.

Reply:
Will inline function in header file help? Then it won't rely on LTO.

Reply:
Yes that works

Reply:
Tested the shared library without LTO cases, does not make much difference still.